JASP-Desktop/widgets/listmodel.cpp

```diff
@@ -122,7 +122,8 @@ Terms ListModel::getSourceTerms()
 			Terms sourceTerms = sourceModel->terms(sourceItem->modelUse);
 
 			for (const QMLListView::SourceType& discardModel : sourceItem->discardModels)
-				sourceTerms.discardWhatDoesContainTheseComponents(discardModel.model->terms(discardModel.modelUse));
+				if (discardModel.model)
```

Comment:
does it make sense to have a `discardModel` in a property called `discardModels` that does not actually have a model?

If so: this is a good fix
If not: why is that discardModel in the list at all?

Reply:
Don't understand your remark...
SourceType is a struct with a model property (of ListModel type) and a discardModels property, which is self a vector of SourceType. There is no discardModel property, it is in the code above just a variable name.

Reply:
What I mean is that `sourceItem->discardModels` is a property, a vector right?

So why is there something in there without a model? Could it just be left out of that vector instead of adding those `if`s everywhere maybe?
Because that seems like something that will be forgotten at some point, or overlooked.

Reply:
A SourceType is structure with the definition of a model (the source property in QML). So it is first set with this definition. Afterwards, when the form is completely read, the real model is searched in the form. If it finds it, it is set to the SourceType.model property, if not, an error message is generated in the form.
The function above tries to read the terms of a source: if a discard model does not exist, this function crashes, that's why I have added this if statement.
It is indeed a quite peculiar situation, but JASP should not crash.

Reply:
And `SourceType` is your own struct right?

Wouldn't it make more sense then to make a `getModelComponents()` function that does the check and returns something useful. 

I really think that needing to check if some property is null in quite a different place then where the struct is defined is bad programming practice to be honest. It is easily forgotten later on, when somebody needs to change the code.

And now if you would need components from the SourceType you can just ask for it instead of needing to check random stuff that is only clear because it crashed somewhere sometime.